src/xunit.analyzers/InlineDataMustMatchTheoryParameters.cs

```diff
@@ -78,52 +86,77 @@ public override void Initialize(AnalysisContext context)
                                 builder.ToImmutable()));
                         }
 
-                        for (int i = 0; i < Math.Min(values.Length, method.Parameters.Length); i++)
+                        int valueIdx = 0, paramIdx = 0;
+                        for (; valueIdx < values.Length && paramIdx < method.Parameters.Length; valueIdx++)
                         {
-                            var parameter = method.Parameters[i];
+                            var parameter = method.Parameters[paramIdx];
+
+                            if (xunitSupportsParameterArrays
+                                && parameter.IsParams
+                                && parameter is IArrayTypeSymbol arrayParameter
+                                && arrayParameter.ElementType == compilation.ObjectType)
+                            {
+                                // Everything is assignable to 'params object[]' so move one
+                                continue;
+                            }
                             if (parameter.Type == compilation.ObjectType)
-                                continue; // Everything is assignable to object so move one
```

Comment:
I think this was supposed to be "move on", so please fix the typo in the copied comment